src/prevector.h

```diff
@@ -314,16 +329,20 @@ class prevector {
     }
 
     void resize(size_type new_size) {
-        if (size() > new_size) {
+        size_type cur_size = size();
+        if (cur_size == new_size) {
+            return;
+        }
+        if (cur_size > new_size) {
             erase(item_ptr(new_size), end());
+            return;
         }
         if (new_size > capacity()) {
             change_capacity(new_size);
         }
-        while (size() < new_size) {
-            _size++;
-            new(static_cast<void*>(item_ptr(size() - 1))) T();
-        }
+        size_t increase = new_size - cur_size;
```

Comment:
Maybe use `ptrdiff_t` here and in `fill`? Would allow you to assert against the negative case rather than have a very large value.